### ISSUE TITLE
Left clicking the panel icon brings up the context menu instead of the main window for autokey-gtk

### ISSUE THREAD

**Issue by ineuw** [autokey/autokey#416]
Since Autokey's menu is the same with the right or the left mouse click, would you consider changing either click to show the main window directly and bypass the current menu.

**Comment by josephj11** [comment]
I don't understand this request. When I left click my panel icon, the AutoKey Main menu opens. When I right click it, I get the context menu. Does it work differently for you?

**Comment by ineuw** [comment]
Ahhhhhhhhhhhhhh, Just remembered that I am using nemonein's 
ppa https://launchpad.net/~nemonein/+archive/ubuntu/autokey.
In this case I know what to do and will post the results.

**Comment by ineuw** [comment]
Uninstalled nemonein's  autokey-common and autokey-gtk. 
Deleted his PPA which seems dormant since February. 
Cleaned everything and rebooted.
Downloaded and installed the 64bit .deb filesfrom https://github.com/autokey/autokey/releases/tag/v0.95.10 
Rebooted but the left and right menu are the same. Only secondary access to the Main menu.
No big deal. I am sure the mystery will be solved

**Comment by josephj11** [comment]
I changed this from an enhancement request to a bug report. Your panel icon is not working correctly. It has worked correctly for me for many releases.
It would be interesting to know if this occurs in autokey-qt as well.
I don't know if running a trace would show anything, but it's worth a try.

**Comment by ineuw** [comment]
Are you referring to debugging? I could do that, and the install the qt version just to try. This panel issue, and the wrong icon is a mystery. Could this come from an old source stored somewhere? Which version is the Autokey that's offered from the software repository of debian/ubuntu/mint? I am in Windows at the moment, and won't be online for several hours.

**Comment by ineuw** [comment]
Installed qt and the main menu appears when the left mouse key is clicked. I will stick to qt. AFAIK there is no difference in the execution of my scripts.

**Comment by josephj11** [comment]
I use the Qt version because I use KDE and have all the libraries anyway.

autokey-common contains all the processing stuff and is used by both versions.

If you're using Debian testing or Ubuntu 20.04, then you get the latest version of AutoKey. Previous versions had 0.90.4 because we didn't have a Debian maintainer to update their package.

I was suggesting starting a trace with `autokey-gtk --verbose` and left clicking the panel icon to see if the trace said anything useful.

**Comment by ineuw** [comment]
Hi Joe, I used --verbose on qt since it was installed. The first output is from ~20 hours ago and another a few minutes ago. Compressed the two files to 16kb zip files. Would someone be interested to look at them, If so, where can I upload it? I will now remove this and re-install autokey-gtk.

**Comment by ineuw** [comment]
Autokey-gtk has the same menu problem as before. Nothing changed. Just add it to a wish list of sorts. It's not important and it will be corrected in due time.

**Comment by josephj11** [comment]
We don't have our own file server, but there are many file sharing services on the Internet. I used Dropbox which gives me 2GB free per email address/account. Google Drive is more generous. Once you upload them somewhere, you will be able to generate a link to them that you can post here.

Since this appears to be a real bug for you, I'm unclear on why you closed this issue.

While we would appreciate an issue filer to stick with the issue to help resolve it, that is certainly not a requirement.

**Comment by ineuw** [comment]
Hi, and sorry about the delay. I will set up the Google Drive's public folder and upload the trace. Had to reinstall Linux Mint today and reinstalled Autokey-gtk and it's still the same. I will create a new trace.

I wonder who else uses gtk and how is the menu working? Should I ask in the Google forum, since more people read that than this.

I closed the issue because I lived with this "bug" since day one, and never knew any other way. So, it's no big deal. It's not worth for the developers to bother with this specifically, and I don't like to create unnecessary work. I have no doubt that they'll come across this issue and correct it (if it's not my configuration that causes this).

BTW, I always install from [(https://github.com/autokey/autokey/releases/tag/v0.95.10)]. using the .deb packages.

**Comment by josephj11** [comment]
ineuw sent me some traces from running the GTK and Qt versions on his system.
Below are snips and my comments from those. A number of things don't look right to me, but I'm not sure what's going on. I have the full traces if they are needed.

The GTK trace looks moderately strange because:

1) It keeps grabbing keys way more than usual. I thought it did all of that at the beginning and then was done with it, but yours are interspersed with things that look like you pressing keys. I thought Thomas suppressed most of those messages, but apparently not.

2) It looks like you pressed a whole lot more keys than just right clicking on the panel icon. I didn't even see where that happened

3) There's an exception near the start of the GTK trace that looks like a real problem

The first Qt trace actually tells us that it is displaying the AutoKey main menu. The other two traces do not.

I have no idea what all the X protocol errors mean.

----

```
2020-05-15 22:00:25,668 ERROR - interface - Got BadWindow error while requesting window information.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/interface.py", line 1074, in get_window_info
    return self._get_window_info(window, traverse)
  File "/usr/lib/python3/dist-packages/autokey/interface.py", line 1080, in _get_window_info
    new_wm_title = self._try_get_window_title(window)
  File "/usr/lib/python3/dist-packages/autokey/interface.py", line 1128, in _try_get_window_title
    atom = self._try_read_property(window, self.__VisibleNameAtom)
  File "/usr/lib/python3/dist-packages/autokey/interface.py", line 1145, in _try_read_property
    return window.get_property(property_name, 0, 0, 255)
  File "/usr/lib/python3/dist-packages/Xlib/xobject/drawable.py", line 461, in get_property
    long_length = length)
  File "/usr/lib/python3/dist-packages/Xlib/protocol/rq.py", line 1369, in __init__
    self.reply()
  File "/usr/lib/python3/dist-packages/Xlib/protocol/rq.py", line 1389, in reply
    raise self._error
Xlib.error.BadWindow: <class 'Xlib.error.BadWindow'>: code = 3, resource_id = <class 'Xlib.xobject.resource.Resource'>(0x05c000ac), sequence_number = 12451, major_opcode = 20, minor_opcode = 0

----

Qt trace 200515-0207

X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 1241, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 1242, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 1243, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 1244, major_opcode = 33, minor_opcode = 0
2020-05-15 02:42:21,637 DEBUG - interface - Grabbing hotkey: ['<super>'] 'p'
2020-05-15 02:42:21,638 DEBUG - interface - Grabbing hotkey: ['<super>'] 'p'
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 1251, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 1252, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 1253, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 1254, major_opcode = 33, minor_opcode = 0

Then

X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 2020, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 2021, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 2022, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 60817411, sequence_number = 2023, major_opcode = 33, minor_opcode = 0
2020-05-15 02:42:21,800 DEBUG - interface - Grabbing hotkey: ['<super>'] 'o'
2020-05-15 02:42:21,801 DEBUG - interface - Grabbing hotkey: ['<super>'] 'o'
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 2030, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 2031, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 2032, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54528281, sequence_number = 2033, major_opcode = 33, minor_opcode = 0

2020-05-15 02:43:26,034 DEBUG - root.Qt-GUI.System-tray-notifier - Triggered system tray icon with reason: 3
2020-05-15 02:43:26,035 INFO - root - Displaying configuration window

Then

X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11295, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11296, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11297, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11298, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11303, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11304, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11305, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 65011736, sequence_number = 11306, major_opcode = 33, minor_opcode = 0

----

Qt trace 200515-0208

X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1196, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1197, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1198, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1199, major_opcode = 33, minor_opcode = 0
2020-05-15 21:04:28,418 DEBUG - interface - Grabbing hotkey: ['<super>'] 'p'
2020-05-15 21:04:28,419 DEBUG - interface - Grabbing hotkey: ['<super>'] 'p'
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1206, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1207, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1208, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1209, major_opcode = 33, minor_opcode = 0

Then

X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1940, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1941, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1942, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 62914563, sequence_number = 1943, major_opcode = 33, minor_opcode = 0
2020-05-15 21:04:28,612 DEBUG - interface - Grabbing hotkey: ['<super>'] 'o'
2020-05-15 21:04:28,613 DEBUG - interface - Grabbing hotkey: ['<super>'] 'o'
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1950, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1951, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1952, major_opcode = 33, minor_opcode = 0
X protocol error:
<class 'Xlib.error.BadAccess'>: code = 10, resource_id = 54526433, sequence_number = 1953, major_opcode = 33, minor_opcode = 0
```

**Comment by ineuw** [comment]
Didn't touch the keyboard. All I did was is is open the terminal and type 'autokey-gtk --verbose'

**Comment by josephj11** [comment]
That's really weird. The whole point of the trace was to see what happened when you left clicked on the panel icon and the wrong display was presented.
We'll have to see what the devs think is going on.

**Comment by ineuw** [comment]
@josephj11 Started Autokey again with --verbose in the terminal, and waited until the trace output stopped.
Then,  clicked left mouse control, waited till the trace output stopped.
Then, cllicked right mouse control, waited till the trace output stopped, 
Then quit the app.

Pasted the results to pastebin at [](https://pastebin.com/wjJWdXQQ) and attached it here a zipped copy as well.
[200523-0127_autokey_qt_verbose.txt.zip](https://github.com/autokey/autokey/files/4671205/200523-0127_autokey_qt_verbose.txt.zip)

**Comment by josephj11** [comment]
That has a lot  more X  errors in it than any of the previous traces.
The trace doesn't say anything about trying to open the Main window.

When you pressed the mouse buttons, was the mouse cursor over the panel icon?

I thought the Qt version worked for you and the problem one was GTK.

What distro are you using? I don't remember.

I'd ask what version of X11 you are using, but I looked on my system and wasn't sure how to determine that.

We'll have to wait for the developers/ input to get any further with this.

**Comment by ineuw** [comment]
The mouse pointer was on the icon in both cases but couldn't take a snapshot.
Copied the important info from Inxi

Cinnamon 4.4.8 Distro: Linux Mint 19.3 Tricia - base: Ubuntu 18.04 bionic

Display: x11 server: X.Org 1.20.5 driver: modesetting unloaded: fbdev,vesa
resolution: 1664x936_59.00~59Hz

OpenGL: renderer: Mesa DRI Intel UHD Graphics 630 (Coffeelake 3x8 GT2) v:
4.5 Mesa 19.2.8 direct render: Yes

**Comment by Elliria** [comment]
Hey there, I see the same thing when i run the GTK version of AutoKey version 0.95.10 in a Kubuntu 20.04 LTS virtual machine. Regardless of whether I click the left or right mouse button on the panel icon, I get the context menu.

**Comment by ineuw** [comment]
Much thanks for your reply. It goes a step further in establishing that it's not my doing. I am pretty sure that this issue is universal.

**Comment by Elliria** [comment]
Any time. And no, it's not your doing. In fact, it's been happening for a long time. I currently have AutoKey v0.90.4 installed in Ubuntu MATE 18.04 LTS (haven't quite updated yet - still deciding which direction to go in) and it exhibits the same behavior.

**Comment by ineuw** [comment]
I used qt for awhile and everything seems to work. Also, I wanted to learn Python and 3+ seemed to me the logical choice.

**Comment by Elliria** [comment]
Yep, the Qt version behaves as expected, bringing up the main window on left-click and the context menu on right-click. And, yes, 3+ is a good choice, but I suppose any Python learning is a good thing.

**Comment by ineuw** [comment]
I am revisiting this issue because it still exists in v16.beta10. I am referring to the autokey-gtk panel menu which still displays the 
same menu with the left or right mouse button.

autokey-gtk --verbose of v16.10beta produced this last line of thousands of the same X protocol error, followed by a summary:

```
X protocol error:
<class 'Xlib.error.BadWindow'>: code = 3, resource_id = <class 'Xlib.xobject.resource.Resource'>(0x034001d7), sequence_number = 21005, major_opcode = 33, minor_opcode = 0
2022-03-19 13:40:47,244 DEBUG - autokey.iomediator.iomediator - Waiting for IoMediator thread to end
2022-03-19 13:40:47,244 DEBUG - autokey.iomediator.iomediator - IoMediator shutdown completed
2022-03-19 13:40:47,244 INFO - autokey.configmanager.configmanager - Persisting configuration
2022-03-19 13:40:47,244 INFO - autokey.configmanager.configmanager - Backing up existing config file
2022-03-19 13:40:47,244 DEBUG - autokey.monitor - Reporting IN_MODIFY event at /home/ineuw/.config/autokey/autokey.json~
2022-03-19 13:40:47,245 DEBUG - autokey.monitor - Reporting IN_MODIFY event at /home/ineuw/.config/autokey/autokey.json
2022-03-19 13:40:47,245 INFO - autokey.configmanager.configmanager - Finished persisting configuration - no errors
2022-03-19 13:40:47,245 DEBUG - autokey.monitor - Reporting IN_MODIFY event at /home/ineuw/.config/autokey/autokey.json
2022-03-19 13:40:47,245 DEBUG - autokey.service - Service shutdown completed.
2022-03-19 13:40:48,246 INFO - autokey.monitor - Shutting down file monitor
2022-03-19 13:40:48,255 DEBUG - autokey.gtkapp - All shutdown tasks complete... quitting
ineuw@LMC203DT:~$ 
```

**Comment by josephj11** [comment]
The x protocol error message looks familiar. I think it had something to do with upgrading from X11...19 to X11...20.
The rest look like normal AutoKey messages issued during shutdown.

Searching issues for badwindow reveals 9 open and 12 closed AutoKey issues.

**Comment by ineuw** [comment]
@josephj11. I didn't find the solution to the menu yet, but the X protocol error is resolved. These errors indicate that some of my Autokey shortcut assignments are already assigned to another app, which turned out to be the LMC access key assignments. I cleared all except <alt-tab> which I use.  Rebooted and the Xprotocol errors were gone.

**Comment by josephj11** [comment]
Cool. What's "LMC"? And, how did you diagnose the key assignment conflicts?

Perfect example of useless error message text that gives you no clue as to what's actually wrong.

**Comment by ineuw** [comment]
I searched for the error text, and came across a post with the same error in Autokey.  A reply stated that it means that there is a collision with an earlier assignment by another app. 

I suspected the Linux Mint Cinnamon (LMC) OS keyboard assignments because I disable all other shortcut keys, or quit Autokey when I can't. 

So, I cleared all OS assignments, rebooted and immediately ran 'autokey-gtk --verbose' which is clean.
